### ISSUE TITLE
Broken relationship between ol.layer.Layer and ol.layer.Vector

### ISSUE THREAD

**Issue by gkresic** [TDesjardins/gwt-ol#154]
I'm not sure if this is gwt-ol's issue or GWT's, but this code throws `ClassCastException`:
```
Map<ol.Feature, ol.layer.Layer> features = new LinkedHashMap<>();
getMap().forEachFeatureAtPixel(
	event.getPixel(),
	(feature, layer) -> {
		features.put(feature, layer);				
		return false;
	}
);
for (Map.Entry<ol.Feature, ol.layer.Layer> entry : features.entrySet()) {
	ol.Feature feature = entry.getKey();
	ol.layer.Layer layer = entry.getValue();	// ClassCastException
}
```

However, if I use `ol.layer.Vector` instead of `ol.layer.Layer` it works:

```
Map<ol.Feature, ol.layer.Vector> features = new LinkedHashMap<>();
getMap().forEachFeatureAtPixel(
	event.getPixel(),
	(feature, layer) -> {
		features.put(feature, (ol.layer.Vector) layer);				
		return false;
	}
);
for (Map.Entry<ol.Feature, ol.layer.Vector> entry : features.entrySet()) {
	ol.Feature feature = entry.getKey();
	ol.layer.Vector layer = entry.getValue();	// works
}
```

I also noticed that if I have some `ol.layer.Vector` instance, like:
```
ol.layer.Vector layer = new ol.layer.Vector();
```
then
```
layer instanceof ol.layer.Vector
```
evaluates (correctly) to `true`, but
```
layer instanceof ol.layer.Layer
```
evaluates to `undefined`.

**Comment by TDesjardins** [comment]
> I also noticed that if I have some `ol.layer.Vector` instance, like:
> 
> ```
> ol.layer.Vector layer = new ol.layer.Vector();
> ```
> 
> then
> 
> ```
> layer instanceof ol.layer.Vector
> ```
> 
> evaluates (correctly) to `true`, but
> 
> ```
> layer instanceof ol.layer.Layer
> ```
> 
> evaluates to `undefined`.

I can't reproduce this behaviour. I added a check in the Unit-Tests which not fails (checked also local with non param constructor):

https://github.com/TDesjardins/gwt-ol/commit/4be2704bc142e20044f3febdbd77ebac15864253
https://travis-ci.org/github/TDesjardins/gwt-ol/jobs/694342319#L1483

You got this issue with GWT 2.8.2 ?

**Comment by gkresic** [comment]
> You got this issue with GWT 2.8.2 ?

Yes, both GWT 2.8.2 and GWT 2.9. Note that I'm testing in "real" browsers using SuperDevMode, not inside GWTTestCase - may it be differences between HtmlUnit and current evergreen browsers?

Code tested:

```
Log.info("Window.Navigator.getPlatform(): " + Window.Navigator.getPlatform());
Log.info("Window.Navigator.getUserAgent(): " + Window.Navigator.getUserAgent());
Log.info("GWT.getVersion(): " + GWT.getVersion());
ol.layer.Vector layer = new ol.layer.Vector();
Log.info("layer instanceof ol.layer.Vector -> " + (layer instanceof ol.layer.Vector));
Log.info("layer instanceof ol.layer.Layer  -> " + (layer instanceof ol.layer.Layer));
```

Platform: Linux 64 bit
Chrome 84 (beta)
Firefox 76.0.1

Outputs:

```
Window.Navigator.getPlatform(): Linux x86_64
Window.Navigator.getUserAgent(): Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.30 Safari/537.36
GWT.getVersion(): 2.9.0
layer instanceof ol.layer.Vector -> true
layer instanceof ol.layer.Layer  -> undefined
```
```
Window.Navigator.getPlatform(): Linux x86_64
Window.Navigator.getUserAgent(): Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0
GWT.getVersion(): 2.9.0
layer instanceof ol.layer.Vector -> true
layer instanceof ol.layer.Layer  -> undefined
```
```
Window.Navigator.getPlatform(): Linux x86_64
Window.Navigator.getUserAgent(): Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.30 Safari/537.36
GWT.getVersion(): 2.8.2
layer instanceof ol.layer.Vector -> true
layer instanceof ol.layer.Layer  -> undefined
```
```
Window.Navigator.getPlatform(): Linux x86_64
Window.Navigator.getUserAgent(): Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0
GWT.getVersion(): 2.8.2
layer instanceof ol.layer.Vector -> true
layer instanceof ol.layer.Layer  -> undefined
```

**Comment by TDesjardins** [comment]
@gkresic Thanks for this nice description! Indeed I can reproduce this behaviour and after some investigations I came to the conclusion that this is a OpenLayers bug that manifests between versions v5.1.3 and v6.0.0-beta.8.

The `@api` annotation for `ol.layer.Layer` was removed with this [commit](https://github.com/openlayers/openlayers/commit/bbe0a66d0730601f455321dcca2af8e0687a17f3#diff-ee7a24cfe5e0ef490b86118796ae14edL65) without putting this to class desc as actually intended.
Only with this [commit](https://github.com/openlayers/openlayers/commit/4a1cda715db97c9a36865d5e804e38458601f39a#diff-ee7a24cfe5e0ef490b86118796ae14edR75) published with version v6.0.0-beta.8 this is fixed.
I can confirm that it works with newer versions of OpenLayers and gwt-ol tested with branch [ol6](https://github.com/TDesjardins/gwt-ol/tree/ol6), so I will close this.

A little bit annoying is the fact that Unit-tests with HTMLUnit doesn't fail in this case. May be I am going to switch test infrastructure on basis of J2Cl.
